functions/HttpTriggerAPIUsers/__init__.py

```diff
@@ -1,52 +1,136 @@
 import json
-import pypyodbc
+import pyodbc
 import logging
 import os
 import azure.functions as func
 
 def main(req: func.HttpRequest) -> func.HttpResponse:
-    logging.info('Python HTTP Database trigger function processed a request.')
-    
+    logging.info(
+        'Python HTTP trigger for /users function is processing a request.')
+
+    # Check request method
+    method = req.method
+    if not method:
+        logging.critical('No method available')
+        raise Exception('No method passed')
+
+    # Create a new connection
+    logging.debug("Attempting DB connection!")
+    try:
+        conn = get_db_connection()
+    except (pyodbc.DatabaseError, pyodbc.InterfaceError) as e:
+        logging.error("Failed to connect to DB: " + e.args[0])
+        logging.debug("Error: " + e.args[1])
```

Comment:
Would logging.critical work better for 500 status codes?

Reply:
Good question! I think it's kind of a toss up. According to Toddy's article on log levels, the critical level is when the application cannot recover and must exit, whereas the error level is when the condition is noticeable by the user and impacts the normal execution of the application. After reading a bit more, I _think_ that error might be the correct level because the function can still be called again and hasn't really crashed, **BUT** I'm totally open to alternative opinions on the matter.

Reply:
I'm not exactly sure, but I don't think it being a 500 has a ton of bearing on the log level, but I could be wrong.

Reply:
In the example with the connection to the DB and the 500 status code, I would agree with Natalia to change it to critical since the application can not continue.  logging.critical would be more appropriate for this 500 status code